src/ui/Search/searchinfilesworker.cpp

```diff
@@ -143,71 +96,158 @@ void SearchInFilesWorker::stop()
-    //int firstLinePosEnd = content->indexOf(newLine, capturedPosStart);
-
-    // Position (from byte 0) of the start of the last line of the found word
-    int lastLinePosStart = content->lastIndexOf(newLine, capturedPosEnd - 1) + 1;
+    FileSearchResult::FileResult fileResult;
+
+    QTextStream stream (content);
+    QString line;
+
+    int i = 0;
+    int column;
+    int streamPosition = 0;
+    int lineLength;
+
+    fileResult.fileName = fileName;
+    if (m_searchMode == SearchHelpers::SearchMode::PlainText) {
+        Qt::CaseSensitivity caseSensitive = m_searchOptions.MatchCase ? Qt::CaseSensitive : Qt::CaseInsensitive;
+        int matchLength = m_string.length();
+
+        while (!(line=stream.readLine()).isNull() && !m_stop) {
+            bool hasResult = true;
+            lineLength = line.length();
+            column = line.indexOf(m_string, 0, caseSensitive);
+            while (column != -1 && !m_stop) {
+                if (m_searchOptions.MatchWholeWord) hasResult = matchesWholeWord(column, matchLength, line);
+                if (hasResult) fileResult.results.append(buildResult(i, column, streamPosition + column, line, matchLength));
+                column = line.indexOf(m_string, column + matchLength, caseSensitive);
+                m_matchCount++;
+            }
+            i++;
+            if (content->midRef(streamPosition + lineLength, 2) == "\r\n") streamPosition += lineLength + 2;
+            else streamPosition += lineLength + 1;
+        }
 
-    // Position (from byte 0) of the end of the last line of the found word
-    int lastLinePosEnd = content->indexOf(newLine, capturedPosStart);
+    }else {
+        QRegularExpressionMatch match;
+        while (!(line=stream.readLine()).isNull() && !m_stop) {
+            lineLength = line.length();
+            match = m_regex.match(line);
+            column = match.capturedStart();
+            while (column != -1 && match.hasMatch() && !m_stop) {
+                fileResult.results.append(buildResult(i, column, streamPosition + column, line, match.capturedLength()));
+                match = m_regex.match(line, column + match.capturedLength());
+                column = match.capturedStart();
+                m_matchCount++;
+            }
+            i++;
+            if (content->midRef(streamPosition + lineLength, 2) == "\r\n") streamPosition += lineLength + 2;
+            else streamPosition += lineLength + 1; 
+        }
+    }
+    return fileResult;
 
-    // String composed by all the lines that contain the found word.
-    QString previewString = content->mid(firstLinePosStart, lastLinePosEnd - firstLinePosStart);
+}
 
-    // All the lines in wholeLine
-    QStringList matchLines = previewString.split(newLine, QString::KeepEmptyParts);
+FileSearchResult::FileResult SearchInFilesWorker::searchMultiLineRegExp(const QString &fileName, QString content)
+{
+    FileSearchResult::FileResult structFileResult;
+    structFileResult.fileName = fileName;
```

Comment:
Why this whole work? Wasn't the flag `QRegularExpression::MultilineOption` sufficient for multiline regexps?